src/Qir/Execution/CommandLineCompiler/Program.cs

```diff
@@ -0,0 +1,146 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.CommandLine;
+using System.CommandLine.Builder;
+using System.CommandLine.Help;
+using System.CommandLine.Invocation;
+using System.CommandLine.Parsing;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Quantum.Qir.Tools;
+
+namespace CommandLineCompiler
+{
+    class Program
+    {
+        private static async Task<int> Main(string[] args)
+        {
+            var buildCommand = CreateBuildCommand();
+
+            var root = new RootCommand() { buildCommand };
+            SetSubCommandAsDefault(root, buildCommand);
+            root.Description = "Command-line tool for processing QIR DLL files.";
+            root.TreatUnmatchedTokensAsErrors = true;
+
+            Console.OutputEncoding = Encoding.UTF8;
+            return await new CommandLineBuilder(root)
+                .UseDefaults()
+                .UseHelpBuilder(context => new QsHelpBuilder(context.Console))
+                .Build()
+                .InvokeAsync(args);
+        }
+
+        /// <summary>
+        /// Creates the Build command for the command line compiler, which is used to
+        /// compile the executables from a given QIR DLL.
+        /// </summary>
+        /// <returns>The Build command.</returns>
+        private static Command CreateBuildCommand()
+        {
+            var buildCommand = new Command("build", "(default) Build the executables from a QIR DLL.")
+            {
+                Handler = CommandHandler.Create((BuildOptions settings) =>
+                    QirTools.BuildFromQSharpDll(settings.QsharpDll, settings.LibraryDirectory, settings.IncludeDirectory, settings.ExecutablesDirectory))
+            };
+            buildCommand.TreatUnmatchedTokensAsErrors = true;
+
+            Option<FileInfo> qsharpDllOption = new Option<FileInfo>(
+                aliases: new string[] { "--qsharpDll", "--dll" },
+                description: "The path to the QIR DLL file that is to be compiled.")
+            {
+                Required = true
+            };
+            buildCommand.AddOption(qsharpDllOption);
+
+            Option<DirectoryInfo> libraryDirectory = new Option<DirectoryInfo>(
```

Comment:
What would the default value be for the library directories option if it is not provided? Should the parameter to the `BuildFromQSharpDll` be nullable, as well as the corresponding parameters to subsequent API functions, or should it be default to some other path. Maybe the includeDirectory path could be used as the default value? Since I'm not really familiar with how this is used in practice it is difficult for me to tell what the right behavior is on my own.

Reply:
I am when creating the option setting `Required = false`. In the near future, the `libraryDirectories` parameter for `BuildFromQSharpDll` will also be optional.

Reply:
Talked about this offline. Since the underlying API currently does not support a list of library directories and a list of include directories, it's ok to leave these options as required now.